### PR TITLE
fix: prevent text resize on pointerover

### LINKED ISSUE #835: [Material] TextBox & PasswordBox Size Issue
## Current behavior

When a `TextBox` or a `PasswordBox` using the Material `OutlinedStyle` and has content receives focus, it probably changes some of its template sizes and it makes the content move to the right by a couple pixels.

Animation_53

## Expected behavior

The content should not be moved when those controls receive focus.

## How to reproduce it (as minimally and precisely as possible)

1. Download [SampTextAndPassBox.zip](https://github.com/unoplatform/Uno.Themes/files/9397636/SampTextAndPassBox.zip)
2. Open the solution
3. Start the application
4. Fill the `PasswordBox` and the `TextBox`
5. Click on both of them
6. Notice that the content seems to be moving to the right

## Environment

Nuget Package: Uno.Material.WinUI

Package Version(s): 2.3.0-dev.6

Affected platform(s):

- [x] iOS
- [x] Android
- [ ] ~WebAssembly~
- [ ] ~UWP~
- [x] WinUI
- [ ] ~MacOS~

## Anything else we need to know?

N/A

### PR DESCRIPTION
**Pull request by kazo0**
closes #835

## PR Type

What kind of change does this PR introduce?

- Bugfix

## Description

186019795-bea66fb0-8c74-435a-98cd-fd3ba69dbc4c

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml [hunk 1]
```diff
@@ -362,11 +363,11 @@
 
 								<VisualState x:Name="Focused">
 									<VisualState.Setters>
-										<Setter Target="Root.BorderBrush"
+										<Setter Target="RootBorder.BorderBrush"
 												Value="{StaticResource PrimaryBrush}" />
 										<Setter Target="PlaceholderElement.Foreground"
 												Value="{StaticResource PrimaryBrush}" />
-										<Setter Target="Root.BorderThickness"
+										<Setter Target="RootBorder.BorderThickness"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml:370]: everywhere where you are changing this BorderThickness, you need to adjust the padding accordingly

File: src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml [hunk 2]
```diff
@@ -452,25 +456,25 @@
...
-							</ScrollViewer.RenderTransform>
-						</ScrollViewer>
+								<ScrollViewer.RenderTransform>
+									<CompositeTransform x:Name="ContentTransform" />
+								</ScrollViewer.RenderTransform>
+							</ScrollViewer>
 
-						<TextBlock x:Name="PlaceholderElement"
+							<TextBlock x:Name="PlaceholderElement"
 								   Grid.Column="1"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml:465]: please get xaml styler and the indentation fixed

File: src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml [hunk 3]
```diff
@@ -379,11 +379,11 @@
...
 								<VisualState x:Name="Focused">
 									<VisualState.Setters>
-										<Setter Target="Root.BorderBrush"
+										<Setter Target="RootBorder.BorderBrush"
 												Value="{StaticResource PrimaryBrush}" />
 										<Setter Target="PlaceholderElement.Foreground"
 												Value="{StaticResource PrimaryBrush}" />
-										<Setter Target="Root.BorderThickness"
+										<Setter Target="RootBorder.BorderThickness"
 												Value="2" />
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml:387]: same here; //padding

File: src/library/Uno.Cupertino/Uno.Cupertino.csproj [hunk 4]
```diff
@@ -86,14 +86,14 @@
 			</PropertyGroup>
 
 			<ItemGroup>
-				<PackageReference Include="Uno.UI" Version="4.2.0-dev.622" />
+				<PackageReference Include="Uno.UI" Version="4.5.0-dev.645" />
```
Comment (Xiaoy312) [inline src/library/Uno.Cupertino/Uno.Cupertino.csproj:89]: keep version bump in another commit, unless it directly relates to the change

**Review by Xiaoy312 — COMMENTED** [review]
mostly: focused state padding

**Review by Xiaoy312 — APPROVED** [review]

**Comment by agneszitte-nventive** [comment]
@kazo0 Are the changes still properly reflecting the different visual states for the controls to match the guidelines in the Uno Figma file ?

**Review by agneszitte-nventive — COMMENTED** [review]
@kazo0 Are the changes still properly reflecting the different visual states for the controls to match the guidelines in the Uno Figma file ?
